diff --git a/esm/history.js b/esm/history.js
index df67820fe3eed558c44fca07a82b0cd409d46720..40da28c20268952d1d3d9cc84716cc8d07da010f 100644
--- a/esm/history.js
+++ b/esm/history.js
@@ -290,7 +290,7 @@ function createBrowserHistory(props) {
 
   function handlePopState(event) {
     // Ignore extraneous popstate events in WebKit.
-    if (isExtraneousPopstateEvent(event)) return;
+    if (isExtraneousPopstateEvent(event) || (history.location.pathname === window.location.pathname && history.location.search === window.location.search)) return;
     handlePop(getDOMLocation(event.state));
   }
 
